sqllogictest/src/runner.rs

```diff
@@ -679,7 +687,24 @@ impl<D: AsyncDB, M: MakeConnection<Conn = D>> Runner<D, M> {
         match (record.clone(), self.apply_record(record).await) {
             (_, RecordOutput::Nothing) => {}
             // Tolerate the mismatched return type...
-            (Record::Statement { .. }, RecordOutput::Query { error: None, .. }) => {}
+            (
+                Record::Statement {
+                    sql,
+                    expected_error,
+                    loc,
+                    ..
+                },
+                RecordOutput::Query { error: None, .. },
+            ) => {
+                if expected_error.is_some() {
+                    return Err(TestErrorKind::MismatchPass {
+                        sql,
+                        expected_err: expected_error.unwrap().to_string(),
+                        kind: RecordKind::Query,
+                    }
+                    .at(loc));
+                }
+            }
             (
                 Record::Query {
```

Comment:
I think this branch has the same problem and can be fixed in the same way. Would you mind fixing it together?

Reply:
ok